src/history/hs_rec.c

```diff
@@ -1104,7 +1101,7 @@ __hs_delete_record(
     WT_ERR_NOTFOUND_OK(__wt_curhs_search_near_before(session, r->hs_cursor), true);
     /* It's possible the value in the history store becomes obsolete concurrently. */
     if (ret == WT_NOTFOUND) {
-        WT_ASSERT(session, tombstone != NULL && __wt_txn_upd_visible_all(session, tombstone));
+        WT_ASSERT(session, tombstone == NULL || __wt_txn_upd_visible_all(session, tombstone));
```

Comment:
Can we add a comment to describe the tombstone equals null case?

Reply:
addressed here: [056f24f](https://github.com/wiredtiger/wiredtiger/pull/9009/commits/056f24fecbebc5fb716984db84519168f9bd30a7)